### ISSUE TITLE
Migrate to TS strict mode

### ISSUE THREAD

**Issue by kubk** [terrasoff/ts-mobx-form-state#8]
The project isn't using TypeScript strict mode. TypeScript strict mode is de-facto standard for TS libraries. If you run `npx tsc --init` in a new project the compiler will generate `tsconfig.json` with enabled `strict: true` by default. Lots of libraries like [Mobx](https://github.com/mobxjs/mobx/blob/main/tsconfig.json#L12), [Redux](https://github.com/reduxjs/redux/blob/master/tsconfig.json#L3) or [Vue](https://github.com/vuejs/vue-next/blob/master/tsconfig.json#L11) use strict mode.

**Comment by terrasoff** [comment]
That's true. It was one of the point in [roadmap](https://github.com/terrasoff/ts-mobx-form-state#roadmap)
